### PR TITLE
[dv regr tool] Generate testplan annotated results

### PR DESCRIPTION
**Pull request by sriyerg**
- Updated script to generate a testplan annotated results table
- Additional minor fixes to support the above change

Note the table is generated using a thirld party python library called
'tabulate'. This needs to be installed separately by running
```
pip3 install --user tabulate
```

It has been added to the `python-requirements.txt` file.

Signed-off-by: Srikrishna Iyer <sriyer@google.com>

### CODE REVIEW

**Comment by sriyerg** [comment]
A sample way to run a uart test and the generated output / result table now looks like this:
```console
 ../../../../util/dvsim.py sim_cfg.hjson -i uart_sanity --reseed 1
INFO: [SimCfg] Scratch path: <somepath>/scratch/regr_results.uart.sim.vcs
INFO: [Deploy] [dvsim]: build:
 [default]
INFO: [Deploy] [dvsim]: [000010s] [build]: [.]
INFO: [Deploy] [dvsim]: [000020s] [build]: [.]
INFO: [Deploy] [dvsim]: run:
 [0.uart_sanity]
INFO: [Deploy] [dvsim]: [000029s] [build]: [P]
INFO: [Deploy] [dvsim]: [000030s] [run]: [.]
INFO: [Deploy] [dvsim]: [000040s] [run]: [.]
INFO: [Deploy] [dvsim]: [000050s] [run]: [.]
INFO: [Deploy] [dvsim]: [000060s] [run]: [.]
INFO: [Deploy] [dvsim]: [000062s] [run]: [P]

# UART Regression Results
  Run on Wednesday January 08 2020 11:38:04AM 

## Test Results
| Milestone   | Name                  | Tests                      | Results   |
|-------------|-----------------------|----------------------------|-----------|
| V1          | sanity                | uart_sanity                | 1/1       |
| V1          | csr                   | uart_csr_hw_reset          | 0/1       |
|             |                       | uart_csr_rw                | 0/1       |
|             |                       | uart_csr_bit_bash          | 0/1       |
|             |                       | uart_csr_aliasing          | 0/1       |
| V1          |                       | TOTAL                      | 1/5       |
| V2          | base_random_seq       | uart_tx_rx                 | 0/1       |
| V2          | parity                | uart_sanity                | 1/1       |
|             |                       | uart_tx_rx                 | 0/1       |
| V2          | parity_error          | uart_rx_parity_err         | 0/1       |
|             |                       | uart_intr                  | 0/1       |
| V2          | watermark             | uart_tx_rx                 | 0/1       |
|             |                       | uart_intr                  | 0/1       |
| V2          | fifo_full             | uart_fifo_full             | 0/1       |
| V2          | fifo_overflow         | uart_fifo_overflow         | 0/1       |
| V2          | fifo_reset            | uart_fifo_reset            | 0/1       |
| V2          | rx_frame_err          | uart_intr                  | 0/1       |
| V2          | rx_break_err          | uart_intr                  | 0/1       |
| V2          | rx_timeout            | uart_intr                  | 0/1       |
| V2          | perf                  | uart_perf                  | 0/1       |
| V2          | sys_loopback          | uart_loopback              | 0/1       |
| V2          | line_loopback         | uart_loopback              | 0/1       |
| V2          | rx_noise_filter       | uart_noise_filter          | 0/1       |
| V2          | rx_start_bit_filter   | uart_rx_start_bit_filter   | 0/1       |
| V2          | tx_overide            | uart_tx_ovrd               | 0/1       |
| V2          | rx_oversample         | uart_rx_oversample         | 0/1       |
| V2          | stress_all            | uart_stress_all            | 0/1       |
| V2          | stress_all_with_reset | uart_stress_all_with_reset | 0/1       |
| V2          | intr_test             | uart_intr_test             | 0/1       |
| V2          | oob_addr_access       | uart_tl_errors             | 0/1       |
| V2          | illegal_access        | uart_tl_errors             | 0/1       |
| V2          | outstanding_access    | uart_csr_hw_reset          | 0/1       |
|             |                       | uart_csr_rw                | 0/1       |
|             |                       | uart_csr_aliasing          | 0/1       |
|             |                       | uart_same_csr_outstanding  | 0/1       |
| V2          | partial_access        | uart_csr_hw_reset          | 0/1       |
|             |                       | uart_csr_rw                | 0/1       |
|             |                       | uart_csr_aliasing          | 0/1       |
| V2          |                       | TOTAL                      | 1/32      |
| V3          |                       | TOTAL                      | 0/0       |
| na          |                       | TOTAL                      | 2/37      |
```

File: hw/ip/uart/data/uart_testplan.hjson [hunk 1]
```diff
@@ -189,7 +189,7 @@
       name: stress_all_with_reset
       desc: '''Have random reset in parallel with stress_all and tl_errors sequences'''
       milestone: V2
-      tests: ["uart_stress_all_with_reset"]
+      tests: ["uart_stress_all_with_rand_reset"]
```
Comment (sriyerg) [inline hw/ip/uart/data/uart_testplan.hjson]: Fixed mapping found during our meeting.

**Comment by sriyerg** [comment]
> LGTM, and it will be nice if we can have an option to generate another simple report that only shows the test pass or fail without appending the result to testplan. For example, if we just run one test or two, we may want to know the status without going through the whole list.

Acknowledged. Will add logic that will automatically enable mapping if ALL tests are run.

**Comment by tjaychen** [comment]
thanks @sriyerg. This looks really cool.

**Comment by rasmus-madsen** [comment]
@sriyerg  this looks good. 
I can't run it until I get IT to install the 3rd party library so that will take some time as always

**Pull request merged**